packages/dataviews/src/components/dataviews-picker/index.tsx

```diff
@@ -0,0 +1,267 @@
+	defaultLayouts,
+	selection,
+	onChangeSelection,
+	header,
+	children,
+	config = { perPageSizes: [ 10, 20, 50, 100 ] },
+	empty,
+}: DataViewsPickerProps< Item > ) {
+	const { infiniteScrollHandler } = paginationInfo;
+	const containerRef = useRef< HTMLDivElement | null >( null );
+	const [ containerWidth, setContainerWidth ] = useState( 0 );
+	const resizeObserverRef = useResizeObserver(
+		( resizeObserverEntries: any ) => {
+			setContainerWidth(
+				resizeObserverEntries[ 0 ].borderBoxSize[ 0 ].inlineSize
+			);
+		},
+		{ box: 'border-box' }
+	);
+	const [ openedFilter, setOpenedFilter ] = useState< string | null >( null );
+	function setSelectionWithChange( value: SelectionOrUpdater ) {
+		const newValue =
+			typeof value === 'function' ? value( selection ) : value;
+		if ( onChangeSelection ) {
+			onChangeSelection( newValue );
+		}
+	}
+	const _fields = useMemo( () => normalizeFields( fields ), [ fields ] );
+	const filters = useFilters( _fields, view );
+	const hasPrimaryOrLockedFilters = useMemo(
+		() =>
+			( filters || [] ).some(
+				( filter ) => filter.isPrimary || filter.isLocked
+			),
+		[ filters ]
+	);
+	const [ isShowingFilter, setIsShowingFilter ] = useState< boolean >(
+		hasPrimaryOrLockedFilters
+	);
+
+	useEffect( () => {
+		if ( hasPrimaryOrLockedFilters && ! isShowingFilter ) {
+			setIsShowingFilter( true );
+		}
+	}, [ hasPrimaryOrLockedFilters, isShowingFilter ] );
+
+	// Attach scroll event listener for infinite scroll
+	useEffect( () => {
+		if ( ! view.infiniteScrollEnabled || ! containerRef.current ) {
+			return;
+		}
+
+		const handleScroll = throttle( ( event: unknown ) => {
+			const target = ( event as Event ).target as HTMLElement;
+			const scrollTop = target.scrollTop;
+			const scrollHeight = target.scrollHeight;
+			const clientHeight = target.clientHeight;
+
+			// Check if user has scrolled near the bottom
+			if ( scrollTop + clientHeight >= scrollHeight - 100 ) {
+				infiniteScrollHandler?.();
+			}
+		}, 100 ); // Throttle to 100ms
+
+		const container = containerRef.current;
+		container.addEventListener( 'scroll', handleScroll );
+
+		return () => {
+			container.removeEventListener( 'scroll', handleScroll );
+			handleScroll.cancel(); // Cancel any pending throttled calls
+		};
+	}, [ infiniteScrollHandler, view.infiniteScrollEnabled ] );
+
+	return (
+		<DataViewsContext.Provider
+			value={ {
+				view,
+				onChangeView,
+				fields: _fields,
+				actions,
+				data,
+				isLoading,
+				paginationInfo,
+				isItemClickable,
+				selection,
+				onChangeSelection: setSelectionWithChange,
+				openedFilter,
+				setOpenedFilter,
+				getItemId,
+				getItemLevel,
+				containerWidth,
+				containerRef,
+				resizeObserverRef,
+				defaultLayouts,
+				filters,
+				isShowingFilter,
+				setIsShowingFilter,
+				config,
+				empty,
+				hasInfiniteScrollHandler: !! infiniteScrollHandler,
+				picker: true,
+			} }
+		>
+			<div className="dataviews-wrapper" ref={ containerRef }>
```

Comment:
```suggestion
			<div className="dataviews-picker-wrapper" ref={ containerRef }>
```

I guess?

Reply:
Yeah, this is where it gets interesting as the styles are exactly the same. We could extract a shared component, though it feels like overkill to me.

I've changed the classname and then made the selectors in the stylesheets `.dataview-wrapper, .dataviews-picker-wrapper`. Not sure it's the best approach, but it's easy to change if we need to.